packages/stargate/src/stargateclient.ts

```diff
@@ -268,7 +270,34 @@ export class StargateClient {
     if (this.tmClient) this.tmClient.disconnect();
   }
 
-  public async broadcastTx(tx: Uint8Array): Promise<BroadcastTxResponse> {
+  public async broadcastTx(
+    tx: Uint8Array,
+    timeoutMs = 60_000,
+    pollIntervalMs = 3_000,
+  ): Promise<BroadcastTxResponse> {
+    return new Promise((resolve, reject) => {
+      const rejectTimeout = setTimeout(reject, timeoutMs, `timeout of ${timeoutMs}ms exceeded`);
```

Comment:
I would create a nice error type (like TimeoutError) that apps can handle. It can create the transaction ID, which we can calculate locally from `tx` (hopefully). Then an app can say: _hey, we submitted your transaction XYZ but did not find it on chain yet. You might want to check manually or come back later._ and store the transaction ID in some local storage.

Reply:
The reason I had the original timeout error re-thrown was because it is essentially the "same" error (regardless of whether rpc timed out or polling did) and maybe people already had their own flow of handling that error/exception and didn't want to break that flow with a new exception. YMMV